src/plugins/auto_batch/auto_batch.cpp

```diff
@@ -804,8 +870,28 @@ InferenceEngine::IExecutableNetworkInternal::Ptr AutoBatchInferencePlugin::LoadN
                             << "Auto-batching operates only networks with inputs/outputs batched by 0th dimension";
             }
         }
-        if (!batched_inputs.size())
-            IE_THROW(NotImplemented) << "Auto-batching supports only networks with inputs featuring batched dim!";
+        const auto& results = function->get_results();
+        for (size_t output_id = 0; output_id < results.size(); output_id++) {
+            const auto& output = results[output_id];
+            const auto& shape = output->get_output_partial_shape(0);
+            // check the batch dim: either 0th (and the original batch size of 1) or none
+            if (shape.size() && ov::DimensionTracker::get_label(shape[0])) {
```

Comment:
Please mirror input checks fully. We could have internal dynamism via NonMaxSuppression/NonZero or other ops which could lead to dynamic output.
```suggestion
            if (shape.is_dynamic())
                    IE_THROW(NotImplemented) << "Auto-batching does not support dynamic networks!";
            if (shape.size() && ov::DimensionTracker::get_label(shape[0])) {
```

Reply:
ok, will fix in the https://github.com/openvinotoolkit/openvino/pull/10446